src/main/kotlin/com/coder/gateway/sdk/CoderCLIManager.kt

```diff
@@ -194,16 +195,35 @@ class CoderCLIManager @JvmOverloads constructor(
         }
     }
 
+    /**
+     * Escape a command argument by wrapping it in double quotes and escaping
+     * any double quotes in the argument.  For example, echo "test" becomes
+     * "echo \"test\"".
+     */
+    private fun escape(s: String): String {
+        return "\"" + s.replace("\"", "\\\"") + "\""
+    }
```

Comment:
This smells like we could run into issues with paths containing
- spaces
- backslashes (e.g. Windows)

Have a look at `sshConfigExecEscape` for what we currently do in the CLI:  https://github.com/coder/coder/blob/main/cli/configssh.go#L787-L810

Reply:
Makes sense!

Reply:
Hmm I am getting myself really confused.  Do you know if SSH on Windows uses `\` as an escape?  I feel like it must.

Because if it does, that means we need to escape the `\` as well, otherwise `C:\"hello"` becomes `"C:\\"hello\""` which is not right.

But then to make matters more confusing, this command is eventually passed into `cmd.exe` and it seems the escaping/quoting situation there is even weirder?  I am reading that the way to quote a `"` is with another `"`.  So that might mean the end value should be `"C:\""hello"""`.

But now, if SSH uses `\` as an escape and `cmd.exe` uses `"`, then the final final end value should be `"C:\\""hello""`.  I think.

Hopefully I can test on a Windows machine soon, just need to figure out how to run a VM.  But if you happen to have the answers already, do let me know!

Reply:
Er, it would be `"C:\\\"\"hello\"\"` actually.

Reply:
I think I was overthinking things with the `"""` quoting, no need to do that since we want the command to run just as the user wrote it (so if you type `C:\echo.exe "hello world"` in the settings we should end up executing `cmd.exe /c C:\echo.exe "hello world"`).

So I think escaping the quotes and slashes gives us that (we end up with `--header-command "C:\\hello.exe \"hello world\""`), or at least in Unix-like shells that would work...

Reply:
I think some of the logic in `sshConfigExecEscape` is also specifically for handling how OpenSSH interprets ProxyCommand. I guess we're just executing the command directly here...

Reply:
We do write this as part of `ProxyCommand` so I should definitely make sure to copy the behavior correctly.

Aahhhhh so I looked at the linked issue and re-read the comment and I think I understand now.  SSH treats quotes differently in configs than Unix-like shells.  Anything in double quotes is treated literally which is why there is no need to escape the slashes (so it is more like single quotes in a shell).  And we could always wrap in quotes but that can break Windows so we only do it if we absolutely need to for spaces or tabs.

I will refactor to match the behavior exactly.